sympy/functions/elementary/hyperbolic.py

```diff
@@ -624,6 +625,40 @@ def as_real_imag(self, deep=True, **hints):
         denom = sinh(re)**2 + cos(im)**2
         return (sinh(re)*cosh(re)/denom, sin(im)*cos(im)/denom)
 
+    def _eval_expand_trig(self, **hints):
+        arg = self.args[0]
+        x = None
+        if arg.is_Add or arg.is_Mul:
+            from sympy import symmetric_poly
+            n = len(arg.args)
+            TX = []
+            if arg.is_Add:
+                for x in arg.args:
+                    tx = tanh(x, evaluate=False)._eval_expand_trig()
+                    TX.append(tx)
+            else:
+                coeff, terms = arg.as_coeff_Mul()
+                #double angle formula
+                if coeff.is_Integer and coeff == 2:
+                    tx = tanh(terms, evaluate=False)._eval_expand_trig()
+                    TX.append(tx)
+                    TX.append(tx)
+                #half angle formula
+                elif coeff == (1/2):
+                    cx = cosh(terms, evaluate=False)._eval_expand_trig()
+                    sx = sinh(terms, evaluate=False)._eval_expand_trig()
+                    return (cx-1)/sx
+                else:
+                    return coeff*tanh(terms)
+            Yg = numbered_symbols('Y')
+            Y = [ next(Yg) for i in range(n) ]
+
+            p = [0, 0]
+            for i in range(n + 1):
+                p[1 - i % 2] += symmetric_poly(i, Y)*(1)**((i % 4)//2)
```

Comment:
What does `(1)**((i % 4)//2)` do?

Reply:
Sorry, it was left over from the tan _eval_expand_trig method. I have removed it in the most recent commit.